code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -45,6 +45,10 @@
 					movement_target = null
 					stop_automated_movement = 0
 					break
+			for(var/obj/item/toy/cattoy/T in view(1,src))
+				if (T.cooldown < (world.time - 400))
+					emote("me", 1, "bats \the [T] around with its paw!")
+					T.cooldown = world.time

```

Comment:
1 - The for loop should be inside the timer check, not the other way around, the whole reason for timer is to not check surroundings each life call
2 - Make the loop break after the first find of mouse by putting the "break" word at the end of the if branch, otherwise it will spam as many messages as there are mouses